### PR TITLE
allow printing negative integers in cc_print

### PR DESCRIPTION
**Pull request by thinkingfish**

### CODE REVIEW

File: src/cc_print.c [hunk 1]
```diff
@@ -47,6 +47,22 @@ cc_print_uint64_unsafe(char *buf, uint64_t n)
 }
 
 size_t
+cc_print_int64_unsafe(char *buf, int64_t n)
+{
+    size_t d;
+    uint64_t ab = llabs(n);
```
Comment (seppo0010) [inline src/cc_print.c]: apparently `llabs` returns an `int64_t` and if `n` is `INT64_MIN` the behaviour is undefined... do we want to support that case? http://www.gnu.org/software/libc/manual/html_node/Absolute-Value.html

Some thing happens in the non-unsafe function.

File: src/cc_print.c [hunk 2]
```diff
@@ -62,6 +78,27 @@ cc_print_uint64(char *buf, size_t size, uint64_t n)
...
+
+    d =  digits(ab) + (n < 0);
+    if (size < d) {
+        return 0;
+    }
+
+    if (n < 0) {
```
Comment (seppo0010) [inline src/cc_print.c]: micro optimization
```c
  if (n < 0) {
    *buf++ = '-';
    d--;
  }

  _print_uint64(buf, d, n);
// ...
```

File: src/cc_print.c (hunk 1 above)
Comment (thinkingfish) [reply src/cc_print.c]: Good catch. I'll use a different way getting the absolute.

**Comment by kevyang** [comment]
ship it

**Pull request merged**